tools/autograd/gen_variable_type.py

```diff
@@ -476,9 +529,11 @@ def gen_differentiable_outputs(f: NativeFunction) -> List[DifferentiableOutput]:
     args_with_derivatives = find_args_with_derivatives(differentiable_inputs)
     differentiable_outputs = gen_differentiable_outputs(f)
 
-    requires_derivative = (
-        base_name not in DONT_REQUIRE_DERIVATIVE and name not in DONT_REQUIRE_DERIVATIVE and
-        len(differentiable_inputs) > 0 and len(differentiable_outputs) > 0)
+    undifferentiable = (base_name in DONT_REQUIRE_DERIVATIVE) or (name in DONT_REQUIRE_DERIVATIVE)
```

Comment:
nit: parentheses are not necessary here

Reply:
I feel like it makes this kind of condition where you have to know precedence of special keywords easier to read.
I would prefer to leave it like that if you're fine with it?

Reply:
well flake8 didn't complain so whatever